### PR TITLE
Dataset documentation + fix most sphinx warnings

### LINKED ISSUE #29: Create documentation
I'm thinking about a "read-the-docs" style doc page and adding some rationale, main features, and some examples in the README.md.

### PR DESCRIPTION
**Pull request by pzelasko**
Resolves #29

### CODE REVIEW

File: lhotse/dataset/unsupervised.py [hunk 1]
```diff
@@ -10,7 +10,8 @@
 class UnsupervisedDataset(Dataset):
     """
     Dataset that contains no supervision - it only provides the features extracted from recordings.
-    The returned features are a torch Tensor of shape (T x F).
+    The returned features are a :class:`torch.Tensor` of shape ``(T x F)``, where T is the number of frames,
+    and F is the feature dimension.
```
Comment (danpovey) [inline lhotse/dataset/unsupervised.py:14]: How will you implement collate_fn?

File: lhotse/dataset/speech_recognition.py [hunk 2]
```diff
@@ -14,7 +14,16 @@
 class SpeechRecognitionDataset(Dataset):
     """
     The PyTorch Dataset for the speech recognition task.
-    Contains acoustic features and the corresponding text.
+    Returns a dict of:
```
Comment (danpovey) [inline lhotse/dataset/speech_recognition.py:17]: When naming this, we might want to bear in mind that in future there may be other versions of this.
E.g. we may want to do padding while collating, to ensure even minibatch sizes.  
(I'm thinking that may be easier to do by implementing the data-loader with no underlying Dataset,
just reading the CutSet directly).
Up to you though.

File: lhotse/dataset/unsupervised.py (hunk 1 above)
Comment (pzelasko) [reply lhotse/dataset/unsupervised.py:14]: I'm considering a scenario where a custom collate_fn is not necessary; we'd instead truncate/pad all the samples into a number of duration bins, and implement a [Sampler](https://pytorch.org/docs/stable/data.html?highlight=sampler#torch.utils.data.Sampler) that constructs batches only from same duration samples. Then we'd only need to implement collate_fn if we opt into returning custom type objects from dataset (like graphs); even then, the only modification to the default collate_fn would be "informing it" how to batch these objects together.

Comment (pzelasko) [reply lhotse/dataset/unsupervised.py:14]: "truncate/pad all the samples into a number of duration bins" -> I meant in the CutSet

File: lhotse/dataset/speech_recognition.py (hunk 2 above)
Comment (pzelasko) [reply lhotse/dataset/speech_recognition.py:17]: Yeah, there'll definitely be more versions; we should rename as necessary. Re collating, check out my previous comment.
